poc/flp_pine.py

```diff
@@ -539,5 +535,9 @@ def range_check(dot_prod: Field,
         v.as_unsigned() <= (upper_bound - lower_bound).as_unsigned()
     return (is_in_range, v, u)
 
+def bits(x):
+    """Return the number of bits required to encode `x`."""
+    return math.ceil(math.log2(x + 1))
```

Comment:
An alternative is to use `bit_length()` in python. I learned that from David when I implemented `MultiHotHistogram`: https://github.com/cfrg/draft-irtf-cfrg-vdaf/blob/main/poc/flp_generic.py#L699

Reply:
Done!